source/test_linked_list.cxx

```diff
@@ -0,0 +1,405 @@
+#include "LinkedList.h"
+
+#include <gtest/gtest.h>
+
+#include <cassert>
+#include <cmath>
+#include <string>
+#include <type_traits>
+
+TEST(LinkedListTest, EmptyListIsEmpty)
+{
+  DataStructures::LinkedList<int> empty{};
```

Comment:
It's common in test packages like this to `using namespace DataStructures` so you can avoid quite so much typing. (This flies in unit tests more easily than it does in library/application code because you're usually testing few things at a go.)